### DISCUSSION TITLE
Should we switch from "@types/jest" to the in-built "@jest/globals" of jest?

### DISCUSSION THREAD

**Discussion by UNIDY2002** [react-native-community/discussions-and-proposals#592] [Ideas]
According to [the documentation](https://jestjs.io/docs/getting-started#type-definitions) and https://github.com/facebook/jest/pull/13133, jest is recommending in favor of their own "@jest/globals".

Would it be better to switch from "@types/jest" to "@jest/globals" in the react-native template?

**Comment by kelset** [comment]
~can you try submitting as a PR so that we can try to see how Meta thinks about it? I think it will be a good change 👍~ (check conv with Simen below)

**Comment by SimenB** [comment]
Note that we don't really have a good story if you wanna use globals instead of importing. We use this in our own repo: https://github.com/facebook/jest/tree/main/packages/test-globals

Not sure if something similar would work for RN (I'm guessing not)

**Reply by kelset** [reply]
thanks for chiming in @SimenB 🙏 so you'd say it's better to not go for the proposed approach?

**Reply by SimenB** [reply]
Not necessarily. You could just use an import in the test you ship with the RN template and point to `@types/jest` for folks who prefer the globals.

Something like https://github.com/DefinitelyTyped/DefinitelyTyped/pull/62037 would be awesome, but I'm not sure how feasible that is.

**Reply by UNIDY2002** [reply]
Thank you @SimenB!

If I understood correctly, according to https://github.com/DefinitelyTyped/DefinitelyTyped/pull/62037, `@types/jest` will become a wrapper of `@jest/globals` in future versions. Just for confirmation in case I misunderstood, is it correct that directly importing from `@jest/globals` in the testcases will be a better practice?

The template for RN new projects still implicitly imports the global APIs (see https://github.com/facebook/react-native/blob/main/template/__tests__/App.test.tsx). My proposal is to change it into an explicit import, and do you think this will be a nice change?

**Reply by SimenB** [reply]
>My proposal is to change it into an explicit import, and do you think this will be a nice change?

I think so, yes. My point was more that there needs to be instructions (docs? Code comments?) for people who prefer globals

**Reply by UNIDY2002** [reply]
Thanks! I think it is worth trying submitting a PR.
